CI: TestEgressGatewayManager: cannot lookup policy entry: lookup: key does not exist

## CI failure

https://github.com/cilium/cilium/actions/runs/16044110144/job/45271428406#step:17:16239

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/pkg/egressgateway   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

--- FAIL: TestEgressGatewayManager (0.50s)

    manager_privileged_test.go:433: 
        	Error Trace:	/host/pkg/egressgateway/manager_privileged_test.go:1131
        	            				/host/pkg/egressgateway/manager_privileged_test.go:433
        	Error:      	Received unexpected error:
        	            	cannot lookup policy entry: lookup: key does not exist
        	Test:       	TestEgressGatewayManager
```

Cilium CI dashboard reports these failures occuring in over 5% of test runs for scheduled/push/etc events against the `main` branch in the past week.

@cilium/egress-gateway has this test changed recently? Could you investigate the failures to check whether there is a code bug or testing bug that we can resolve to make the test more reliable?

--- a/pkg/egressgateway/manager_privileged_test.go
+++ b/pkg/egressgateway/manager_privileged_test.go
@@ -403,6 +403,7 @@ func TestEgressGatewayManager(t *testing.T) {
 	// Restore old DestCIDR
 	policy1.destinationCIDRs = []string{destCIDR, destCIDRv6}
 	addPolicy(t, k.policies, &policy1)
+	reconciliationEventsCount = waitForReconciliationRun(t, egressGatewayManager, reconciliationEventsCount)
 
 	// Create a new policy
 	addPolicy(t, k.policies, &policyParams{

